cartographer_grpc/framework/rpc.cc

```diff
@@ -269,10 +220,56 @@ bool* Rpc::GetRpcEventState(Event event) {
       return &read_event_pending_;
     case Event::WRITE:
       return &write_event_pending_;
+    case Event::WRITE_NEEDED:
+      return &write_needed_event_pending_;
   }
   LOG(FATAL) << "Never reached.";
 }
 
+void Rpc::EnqueueMessage(SendItem&& send_item) {
+  cartographer::common::MutexLocker locker(&send_queue_lock_);
+  send_queue_.emplace(SendItem{std::move(send_item.msg), send_item.status});
+}
+
+void Rpc::PerformFinish(std::unique_ptr<::google::protobuf::Message> message,
+                        ::grpc::Status status) {
+  SetRpcEventState(Event::FINISH, true);
+  switch (rpc_handler_info_.rpc_type) {
+    case ::grpc::internal::RpcMethod::BIDI_STREAMING:
+      CHECK(!message);
+      server_async_reader_writer_->Finish(
+          status,
+          new RpcEvent{Event::FINISH, weak_ptr_factory_(this), event_queue_});
```

Comment:
event_queue_ cannot be here, should be a bool

Reply:
Done.